controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -373,6 +373,94 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_search(Socket *sd,
     return son_actions::send_cmdu_to_agent(sd, cmdu_tx);
 }
 
+/**
+ * @brief Encrypt the config data using AES and add to the WSC M2 TLV
+ *        The enctypted data length is the config data length aligned to 16 bytes boundary.
+ *
+ * @param m2 WSC M2 TLV
+ * @param config_data config data in network byte order (swapped)
+ * @return true on success
+ * @return false on failure
+ */
+bool master_thread::autoconfig_wsc_add_m2_encrypted_settings(
+    std::shared_ptr<ieee1905_1::tlvWscM2> m2, WSC::cConfigData &config_data)
+{
+    // Calculate length of data to encrypt
+    // (= plaintext length + 32 bits HMAC aligned to 16 bytes boundary)
+    size_t len             = (config_data.getLen() + 8 + 16) & ~0xFU;
```

Comment:
This formula does not correspond to the comment above...

Based on the comment, I would expect it to be:
```suggestion
    size_t len             = (config_data.getLen() + sizeof(sWscAttrAuthenticator) + 15) & ~0xFU;
```

I don't know which one is correct though.

Also, I think it's worth creating a function somewhere in common to do round_up_to_multiple(). So this would become:
```suggestion
    size_t len             = mapf::round_up_to_multiple(config_data.getLen() + sizeof(sWscAttrAuthenticator, 16);
```

Finally, it would be better to replace the 16 with a constant that is defined in mapf::encryption.

All of the above are probably (nitpick).

Reply:
will be fixed as part of #158